### DISCUSSION TITLE
BlueJay uses phone slot  glitch?

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#2830] [Q&A]
I experienced this problem on Android 11, my mission-mode phone.
I have since recreated the problem on an Android 9 test phone.

1- Set up xDrip (May 3, 2023) to be the collector from a regular G6.  
You should now have readings in xDrip.  Engineering mode is disabled.  So, xDrip is receiving readings from G6 slot 2.

2- Enable BlueJay from xDrip and link with a BlueJay GTS.  
You should now have readings on BlueJay as well.  So, BlueJay is receiving readings from G6 slot 1.

3- Go to xDrip BlueJay Advanced settings page.
Disable "Run Phone Collector".  Enable "BlueJay Uses Phone slot".
You should now have readings on both xDrip and BlueJay.  However, xDrip is only showing readings it receives from BluerJay through Bluetooth.
This means BlueJay is now using slot 2 and xDrip is not directly linked to G6 any longer.

4- After a few readings, go back to xDrip BlueJay Advanced settings page.
Disable "BlueJay Uses Phone slot".  Enable "Run Phone Collector".  So, undo what we did in 3.
xDrip should now become an independent collector again using slot 2.  However, looking at status page, I could see connectivity problems.

5- Power down BlueJay.
xDrip stops working.  No more readings.
I did everything I could think of, like forgetting device, clearing Bluetooth devices on the phone, restarting the phone, etc. to solve the problem with no success.  I even uninstalled xDrip and installed again.  No luck.  I even tried other phones.  None could establish connectivity to G6.  It is like the transmitter slot 2 is not available any longer!

6- To resolve the problem, I enabled engineering mode and changed the slot to 1.
xDrip connected immediately.  After a reading or two, I deleted the slot number from the setting.  The next reading came through, meaning xDrip was working fine back on slot 2.  At this point, I disabled engineering mode.   xDrip continued to work fine.
Powering on BlueJay, it also continued to work fine.

Please do not test this if you are not confident.
Please tell me if you experience the same issue or not.  If you do, please tell me if you can think of a simpler solution.

On a somewhat related note, I would like to remove the ability to change the Bluetooth slot from engineering mode.  This is the PR that does that: https://github.com/NightscoutFoundation/xDrip/pull/2806

**Comment by jamorham** [comment]
I would expect the BlueJay Admin panel `CONFIG` button to solve this as it sounds like a mismatched configuration issue between phone and bluejay.

**Comment by Navid200** [comment]
When the problem manifests itself, in step 5, BluieJay is off.  
Should I tap on Config then when BlueJay is off?
